lib/models/persistent_tab_controller.dart

```diff
@@ -18,7 +25,16 @@ class PersistentTabController extends ChangeNotifier {
       return;
     }
     if (!isUndo) {
-      _tabHistory.add(_index);
+      if (_historyLength > 0) {
+        _tabHistory.add(_index);
+      }
+
+      if (_tabHistory.length > _historyLength) {
+        _tabHistory.removeRange(1, _tabHistory.length - _historyLength + 1);
```

Comment:
I cant think of a case in which it would be required to remove multiple history entries at once. If this had no exact reason I would prefer `_tabHistory.removeAt(1)`